docs/limitations.md

```diff
@@ -0,0 +1,23 @@
+## Limitations
+
+### app name cannot contain uppercase letters
```

Comment:
This explains rkt's limitation about app names. What's explained here is not a limitation anymore since we do the sanitization in the rklet.

However, I think @alban meant that we should explain the limitations of the sanitization we do in the rktlet. That is, if two apps are added to a pod, one `foobar` and another `FooBar`, it will fail because both will be sanitized to the same name (`foobar`).

Reply:
Yes, I meant that. The explanation about the reason is good to have, but as a detail rather than the main point.

Reply:
@iaguis @alban Updated the document. PTAL.